enumerable_methods.rb

```diff
@@ -0,0 +1,63 @@
+# Project 2: Enumerable Methods
+
+module Enumerable
+  def my_each(&proc)
+    length.times do |i|
+      proc.call(self[i])
+    end
+    self
+  end
+
+  def my_each_with_index(&proc)
+    length.times do |item, _i|
+      proc.call(item)
+    end
+    self
+  end
```

Comment:
* [ ] should have working `my_each_with_index` identical and returns the same thing as ruby's [`each_with_index`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-each_with_index). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_each_with_index.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

* [ ] my_each_with_index returns an enumerator if no block is given
  ```ruby
    array = [1,2,5,8,7,98,4]
    p array.my_each_with_index.class # returns this error
    # NoMethodError:
       undefined method `call' for nil:NilClass
     # ./enumerable_methods.rb:13:in `block in my_each_with_index'
     # ./enumerable_methods.rb:12:in `times'
     # ./enumerable_methods.rb:12:in `my_each_with_index'
  ```